src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php

```diff
@@ -615,6 +617,77 @@ public function iShouldNotSeeTheThankYouPage(): void
         Assert::same($this->client->getResponse()->getStatusCode(), 400);
     }
 
+    /**
+     * @Then address to :fullName should be used for both shipping and billing of my order
+     */
+    public function iShouldSeeThisShippingAddressAsShippingAndBillingAddress($fullName): void
+    {
+        $this->iShouldSeeThisShippingAddressAsShippingAddress($fullName);
+        $this->iShouldSeeThisBillingAddressAsBillingAddress($fullName);
+    }
+
+    /**
+     * @Then my order's shipping address should be to :fullName
+     */
+    public function iShouldSeeThisShippingAddressAsShippingAddress(string $fullName): void
+    {
+        /** @var Response $response */
+        $response = $this->client->getResponse();
+        $name = explode(" ",$fullName);
+
+        Assert::same($this->responseChecker->getResponseContent($response)['shippingAddress']['firstName'],
+            $name[0]
+        );
+        Assert::same(
+            $this->responseChecker->getResponseContent($response)['shippingAddress']['lastName'],
+            $name[1]
+        );
+    }
+
+    /**
+     * @Then my order's billing address should be to :fullName
+     */
+    public function iShouldSeeThisBillingAddressAsBillingAddress(string $fullName): void
+    {
+        /** @var Response $response */
+        $response = $this->client->getResponse();
+        $name = explode(" ",$fullName);
+
+        Assert::same($this->responseChecker->getResponseContent($response)['billingAddress']['firstName'],
+            $name[0]
+        );
+        Assert::same(
+            $this->responseChecker->getResponseContent($response)['billingAddress']['lastName'],
+            $name[1]
+        );
```

Comment:
As it looks almost the same as the method above, you could extract a private method for this check with an argument `addressType` or sth like that